packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/ConcatIterator.kt

```diff
@@ -0,0 +1,19 @@
+package expo.modules.kotlin
+
+/**
+ * Simple iterator that will merge two other iterators.
+ */
+class ConcatIterator<T>(
+  private val first: Iterator<T>,
+  private val second: Iterator<T>
+) : Iterator<T> {
+  override fun hasNext(): Boolean = first.hasNext() || second.hasNext()
+
+  override fun next(): T {
+    if (first.hasNext()) {
+      return first.next()
+    }
+
+    return second.next()
```

Comment:
I think I'm obsessed 😅 
```suggestion
    return if (first.hasNext()) {
      first.next()
    } else {
      second.next()
    }
```

![kotlin](https://user-images.githubusercontent.com/278340/170497588-698c86cb-7bc1-476d-bd60-02dcce967b1d.png)

Reply:
```kt
  override fun next() =
    run { first.takeIf { it.hasNext() } ?: second }.next()
```
😆